Source/PagesController.swift

```diff
@@ -172,6 +174,15 @@ extension PagesController : UIPageViewControllerDelegate {
 
 extension PagesController {
 
+  private func setViewController(viewController: UIViewController, atPage page:Int) {
+    pagesDelegate?.pageViewController(self,
+      setViewController: viewController,
+      atPage: page)
+    if let pageChangeHandler = pageChangeHandler {
+      pageChangeHandler(viewController: viewController, page: page)
```

Comment:
@3lvis To make it more clear:
1. We have `PagesController` where is all the logic for page changing.
2. We have `PresentationController` which extends `PagesController` and needs to observe when the page was changed.
3. We have some `ViewController` which deals with `PresentationController` and also needs to be notified about page change.

We can't use delegate, because `pagesDelegate` could be used only for one of these controllers. Subclassing also doesn't work, because we don't want to make `public` all the methods where page could be changed and then override in subclass. So I see 4 workarounds:
1. Add `closure` like I did, then it could be used in the subclass of `PagesController`, in our case - in `PresentationController`, and `pagesDelegate` will be "free" for other use cases.
2. Use KVO in subclass.
3. Use notifications.
4. Implement specific delegate in `PresentationController` (like `presentationDelegate`) and deal with it when we use `PresentationController`. But then we should clarify that setting of `pagesDelegate` will break your `PresentationController`.

Any other thought?